graph/validation_barrier.go

```diff
@@ -106,47 +117,100 @@ func (v *ValidationBarrier) InitJobDependencies(job interface{}) {
-	// type,s and also the types that we use within the database.
 	case *lnwire.ChannelAnnouncement1:
+		id := JobID(v.idCtr.Add(1))
 
-		// We ensure that we only create a new announcement signal iff,
-		// one doesn't already exist, as there may be duplicate
-		// announcements.  We'll close this signal once the
-		// ChannelAnnouncement has been validated. This will result in
-		// all the dependent jobs being unlocked so they can finish
-		// execution themselves.
-		if _, ok := v.chanAnnFinSignal[msg.ShortChannelID]; !ok {
-			// We'll create the channel that we close after we
-			// validate this announcement. All dependants will
-			// point to this same channel, so they'll be unblocked
-			// at the same time.
-			signals := &validationSignals{
-				allow: make(chan struct{}),
-				deny:  make(chan struct{}),
-			}
-
-			v.chanAnnFinSignal[msg.ShortChannelID] = signals
-			v.chanEdgeDependencies[msg.ShortChannelID] = signals
+		v.updateOrCreateJobInfo(msg.ShortChannelID, id)
+		v.updateOrCreateJobInfo(route.Vertex(msg.NodeID1), id)
+		v.updateOrCreateJobInfo(route.Vertex(msg.NodeID2), id)
 
-			v.nodeAnnDependencies[route.Vertex(msg.NodeID1)] = signals
-			v.nodeAnnDependencies[route.Vertex(msg.NodeID2)] = signals
-		}
+		return id, nil
 
-	// These other types don't have any dependants, so no further
-	// initialization needs to be done beyond just occupying a job slot.
+	// Populate the dependency mappings for the below child jobs.
 	case *lnwire.ChannelUpdate1:
-		return
+		childJobID := JobID(v.idCtr.Add(1))
+		v.populateDependencies(childJobID, msg.ShortChannelID)
+
+		return childJobID, nil
 	case *lnwire.NodeAnnouncement:
-		// TODO(roasbeef): node ann needs to wait on existing channel updates
-		return
+		childJobID := JobID(v.idCtr.Add(1))
+		v.populateDependencies(childJobID, route.Vertex(msg.NodeID))
+
+		return childJobID, nil
 	case *lnwire.AnnounceSignatures1:
 		// TODO(roasbeef): need to wait on chan ann?
-		return
+		// - We can do the above by calling populateDependencies. For
+		//   now, while we evaluate potential side effects, don't do
+		//   anything with childJobID and just return it.
+		childJobID := JobID(v.idCtr.Add(1))
+		return childJobID, nil
+
+	default:
+		// An invalid message was passed into InitJobDependencies.
+		// Return an error.
+		return JobID(0), errors.New("invalid message")
+	}
+}
+
+// updateOrCreateJobInfo modifies the set of activeParentJobs for this annID
+// and updates jobInfoMap. This must only be called from InitJobDependencies.
+// NOTE: MUST be called with the mutex held.
+func (v *ValidationBarrier) updateOrCreateJobInfo(annID any, annJobID JobID) {
+	info, ok := v.jobInfoMap[annID]
+	if ok {
+		// If an entry already exists for annID, then a job related to
+		// it is being validated. Add to the set of parent job ids.
+		// This addition will only affect _later_, _child_ jobs for the
+		// annID.
+		info.activeParentJobIDs.Add(annJobID)
+	} else {
```

Comment:
style nit: return after `info.activeParentJobIDs.Add(annJobID)` and remove & unindent the `else` block